rep-2001.rst

```diff
@@ -0,0 +1,138 @@
+REP: 2001
+Title: ROS Bouncy and Newer Variants
+Author: Tully Foote <tfoote@openrobotics.org>, Mikael Arguedas <mikael@openrobotics.org>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 16-Jul-2018
+Post-History:
+
+
+Abstract
+========
+
+This REP describes the variants for the ROS Bouncy release and for releases that follow until otherwise replaced by a similar update REP.
+
+
+Motivation
+==========
+
+For a discussion on the general motivation and role of variants,
+please see REP 108 [1]_.
+
+This document covers the common variants used for recommended
+installations.
+It is expressed in the ros2/variants repository on github.com [2]_.
+
+The definitions in this REP are concise for clearer purpose, and do not list all transitive dependencies explicitly.
+
+
+Specification
+=============
+
+End-user entry points
+---------------------
+
+We define three main entry points for ROS users.
+
+ * desktop (recommended)
+ * ros_base
+ * ros_core
+
+Variants
+--------
+
+ROS Core
+''''''''
+
+The `ros_core` variants composes the core communication protocols.
+It may not contain any GUI dependencies.
+
+::
+
+ - ros_core:
+      packages: [ament_cmake, ament_cmake_ros, ament_index, ament_lint,
+                 ament_package, class_loader, common_interfaces,
+                 console_bridge, googletest, launch, libyaml_vendor,
+                 osrf_pycommon, osrf_testing_tools_cpp, pluginlib,
+                 poco_vendor, rcl, rcl_interfaces, rclcpp, rclpy,
+                 rcutils, rmw, rmw_implementation, ros2cli, rosidl,
+                 rosidl_dds, rosidl_defaults, rosidl_python, rosidl_typesupport,
+                 ros_environment, tinyxml2_vendor, uncrustify]
+      And at least one of the following rmw_implementation:
+      - Fast-RTPS: [Fast-CDR, Fast-RTPS, rmw_fastrtps]
+      - Connext: [rmw_connext, rosidl_typesupport_connext]
+      - Opensplice: [rmw_opensplice, rosidl_typesupport_opensplice]
+
+
+ROS Base
+''''''''
+
+The `ros_base` metapackage composes the `ros_core` metapackage with
+commonly used libraries.
+It may not contain any GUI dependencies.
+
+::
+
+  - ros_base:
+      extends: [ros_core]
+      packages: [geometry2, kdl_parser, orocos_kinematics_dynamics,
+                 robot_state_publisher, tinyxml_vendor, urdf, urdfdom,
+                 urdfdom_headers]
+
+
+Desktop variants
+''''''''''''''''
+
+The `desktop` metapackage is the main entry point for users.
+It provides all commonly used libraries as well as visualization tools and tutorials.
+
+::
+
+  - desktop:
+      extends: [ros_base]
+      packages: [angles, demos, depthimage_to_laserscan, example_interfaces,
```

Comment:
angles used to be below geometry, I think it would make more sense in base with the other geometric libraries.

Reply:
We changed this recently in ROS1 (melodic): https://github.com/ros-infrastructure/rep/pull/144#discussion_r174930746 so I followed the same logic for ROS 2

Reply:
Ok, consistency is good.